sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -0,0 +1,310 @@
+                        cuts = obs.flags.glitch_flags + obs.flags.left_scan
+                    elif split_labels[n_split] == 'scan_right':
+                        cuts = obs.flags.glitch_flags + obs.flags.right_scan
+                    pmap_local = coords.pmat.P.for_tod(obs, comps=self.comps, geom=self.rhs.geometry, rot=rot, threads="domdir", weather=unarr(obs.weather), site=unarr(obs.site), cuts=cuts, hwp=True)
+            else:
+                pmap_local = pmap
+                    
+            # Build the RHS for this observation
+            obs_rhs = pmap_local.zeros() # this is the final RHS, we will fill it at the end
+            
+            if not(self.singlestream):
+                obs_rhs_T = pmap_local.to_map(tod=obs, signal=obs.dsT, comps='T', det_weights=2*nmat.ivar)
+                
+                obs_rhs_demodQ = pmap_local.to_map(tod=obs, signal=obs.demodQ, comps='QU', det_weights=nmat.ivar)
+                obs_rhs_demodU = pmap_local.to_map(tod=obs, signal=obs.demodU, comps='QU', det_weights=nmat.ivar)
+                obs_rhs_demodQU = pmap_local.zeros(super_shape=(2), comps='QU',)
+                
+                obs_rhs_demodQU[0][:] = obs_rhs_demodQ[0] - obs_rhs_demodU[1]
+                obs_rhs_demodQU[1][:] = obs_rhs_demodQ[1] + obs_rhs_demodU[0]
+                del obs_rhs_demodQ, obs_rhs_demodU
+                
+                # we write into the obs_rhs. 
+                obs_rhs[0] = obs_rhs_T[0]
+                obs_rhs[1] = obs_rhs_demodQU[0]
+                obs_rhs[2] = obs_rhs_demodQU[1]
+                del obs_rhs_demodQU, obs_rhs_T
+                                
+                obs_div    = pmap_local.zeros(super_shape=(self.ncomp, self.ncomp))
+                # Build the per-pixel inverse covmat for this observation
+                wT = pmap_local.to_weights(obs, signal=obs.dsT, comps='T', det_weights=2*nmat.ivar,)
+                wQU = pmap_local.to_weights(obs, signal=obs.demodQ, comps='T', det_weights=nmat.ivar,)
+                obs_div[0,0] = wT
+                obs_div[1,1] = wQU
+                obs_div[2,2] = wQU
+                # Build hitcount
+                Nd[0,:] = 1
+                obs_hits = pmap_local.to_map(signal=Nd[0],)
+            else:
+                pmap_local.to_map(dest=obs_rhs, signal=Nd, comps='TQU')
+                obs_div    = pmap_local.zeros(super_shape=(self.ncomp,self.ncomp))
+                pmap_local.to_weights(dest=obs_div, signal=Nd, comps='TQU')
+                Nd[:] = 1
+                obs_hits = pmap_local.to_map(signal=Nd,)
+                
+            # Update our full rhs and div. This works for both plain and distributed maps
+            self.rhs[n_split] = self.rhs[n_split].insert(obs_rhs, op=np.ndarray.__iadd__)
+            self.div[n_split] = self.div[n_split].insert(obs_div, op=np.ndarray.__iadd__)
+            self.hits[n_split] = self.hits[n_split].insert(obs_hits[0],op=np.ndarray.__iadd__)
```

Comment:
I think we want single observation (weight-removed) map here.
Something corresponding to: https://github.com/simonsobs/sotodlib/blob/b40005a53e83208778e978b4339b5719cb6fc09e/sotodlib/coords/demod.py#L136-L138